homeassistant/components/vacuum/__init__.py

```diff
@@ -0,0 +1,360 @@
+    SERVICE_START_PAUSE: {'method': 'async_start_pause'},
+    SERVICE_RETURN_TO_BASE: {'method': 'async_return_to_base'},
+    SERVICE_LOCATE: {'method': 'async_locate'},
+    SERVICE_STOP: {'method': 'async_stop'},
+    SERVICE_SET_FAN_SPEED: {'method': 'async_set_fan_speed',
+                            'schema': VACUUM_SET_FAN_SPEED_SERVICE_SCHEMA},
+    SERVICE_SEND_COMMAND: {'method': 'async_send_command',
+                           'schema': VACUUM_SEND_COMMAND_SERVICE_SCHEMA},
+}
+
+DEFAULT_NAME = 'Vacuum cleaner robot'
+DEFAULT_ICON = 'mdi:google-circles-group'
+
+SUPPORT_TURN_ON = 1
+SUPPORT_TURN_OFF = 2
+SUPPORT_PAUSE = 4
+SUPPORT_STOP = 8
+SUPPORT_RETURN_HOME = 16
+SUPPORT_FAN_SPEED = 32
+SUPPORT_BATTERY = 64
+SUPPORT_STATUS = 128
+SUPPORT_SEND_COMMAND = 256
+SUPPORT_LOCATE = 512
+SUPPORT_MAP = 1024
+
+
+@bind_hass
+def is_on(hass, entity_id=None):
+    """Return if the vacuum is on based on the statemachine."""
+    entity_id = entity_id or ENTITY_ID_ALL_VACUUMS
+    return hass.states.is_state(entity_id, STATE_ON)
+
+
+@bind_hass
+def turn_on(hass, entity_id=None):
+    """Turn all or specified vacuum on."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_TURN_ON, data)
+
+
+@bind_hass
+def turn_off(hass, entity_id=None):
+    """Turn all or specified vacuum off."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_TURN_OFF, data)
+
+
+@bind_hass
+def toggle(hass, entity_id=None):
+    """Toggle all or specified vacuum."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_TOGGLE, data)
+
+
+@bind_hass
+def locate(hass, entity_id=None):
+    """Locate all or specified vacuum."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_LOCATE, data)
+
+
+@bind_hass
+def return_to_base(hass, entity_id=None):
+    """Tell all or specified vacuum to return to base."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_RETURN_TO_BASE, data)
+
+
+@bind_hass
+def start_pause(hass, entity_id=None):
+    """Tell all or specified vacuum to start or pause the current task."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_START_PAUSE, data)
+
+
+@bind_hass
+def stop(hass, entity_id=None):
+    """Stop all or specified vacuum."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_STOP, data)
+
+
+@bind_hass
+def set_fan_speed(hass, fan_speed, entity_id=None):
+    """Set fan speed for all or specified vacuum."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    data[ATTR_FAN_SPEED] = fan_speed
+    hass.services.call(DOMAIN, SERVICE_SET_FAN_SPEED, data)
+
+
+@bind_hass
+def send_command(hass, command, params, entity_id=None):
```

Comment:
Since `params` is optional in the service schema, it should be a keyword argument and default to None here.